deploy/mainnet/464_deploy_CUSDMetaPoolV3.ts

```diff
@@ -0,0 +1,85 @@
+import { HardhatRuntimeEnvironment } from "hardhat/types"
+import { DeployFunction } from "hardhat-deploy/types"
+
+// Deployment names
+const META_POOL_NAME = "SaddleWCUSDMetaPoolV3"
+const META_POOL_LP_TOKEN_NAME = `${META_POOL_NAME}LPToken`
+const BASE_POOL_NAME = "SaddleUSDPoolV2"
+
+// Constructor arguments
+const TOKEN_NAMES = ["WCUSD", `${BASE_POOL_NAME}LPToken`]
+const TOKEN_DECIMALS = [18, 18]
+const LP_TOKEN_NAME = "Saddle wCUSD/saddleUSD-V2"
+const LP_TOKEN_SYMBOL = "saddleWCUSD"
+const INITIAL_A = 100
+const SWAP_FEE = 4e6 // 4bps
+const ADMIN_FEE = 0
+
+const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
+  const { deployments, getNamedAccounts } = hre
+  const { execute, get, getOrNull, log, read, save } = deployments
+  const { deployer } = await getNamedAccounts()
+
+  // Manually check if the pool is already deployed
+  const metaPool = await getOrNull(META_POOL_NAME)
+  if (metaPool) {
+    log(`reusing ${META_POOL_NAME} at ${metaPool.address}`)
+  } else {
+    const TOKEN_ADDRESSES = await Promise.all(
+      TOKEN_NAMES.map(async (name) => (await get(name)).address),
+    )
+
+    const receipt = await execute(
+      "SwapDeployer",
+      {
+        from: deployer,
+        log: true,
+      },
+      "deployMetaSwap",
+      (
+        await get("MetaSwapV3")
+      ).address,
+      TOKEN_ADDRESSES,
+      TOKEN_DECIMALS,
+      LP_TOKEN_NAME,
+      LP_TOKEN_SYMBOL,
+      INITIAL_A,
+      SWAP_FEE,
+      ADMIN_FEE,
+      (
+        await get("LPToken")
+      ).address,
+      (
+        await get(BASE_POOL_NAME)
+      ).address,
+    )
+
+    const newPoolEvent = receipt?.events?.find(
+      (e: any) => e["event"] == "NewSwapPool",
```

Comment:
+1 on Duy's comments. Should never use loose equality in JS. https://2ality.com/2011/12/strict-equality-exemptions.html